src/components/SiteNav/index.astro

```diff
@@ -1,30 +1,67 @@
 ---
 import { Image } from 'astro:assets';
 import logo from '../../assets/icons/logo.svg';
+import closeIcon from './assets/close-icon.svg';
+import hamburgerIcon from './assets/hamburger-menu.svg';
 import './styles.css';
 const BLOG_URL = Astro.site?.href;
 ---
-<nav id="site-nav">
-	<ul>
+
+<nav id="site-nav" aria-label="Main navigation menu">
+	<ul id="nav-menubar">
 		<li>
 			<a href={BLOG_URL}>
 				<Image src={logo} alt="TorontoJS Logo!" height={70} width={70} id="logo-rounded" />
 			</a>
 		</li>
-		<li>
-			<a href={`${BLOG_URL}about`}>
-				About
-			</a>
-		</li>
-		<li>
+		<li class="desktop-only">
 			<a href={`${BLOG_URL}accessibility`}>
 				Accessibility
 			</a>
 		</li>
-		<li>
+		<li class="desktop-only">
 			<a href={`${BLOG_URL}license`}>
 				License
 			</a>
 		</li>
+		<li class="desktop-only">
+			<a href={`${BLOG_URL}about`}>
+				About
+			</a>
+		</li>
+		<li class="mobile-only">
+			<button
+				id="nav-menu-button" aria-controls="mobile-menu" aria-expanded="false" aria-haspopup="true"
+				aria-label="Open blog navigation" popovertarget="mobile-menu"
+			>
+				<Image
+					id="hamburger-icon" src={hamburgerIcon} alt="Hamburger Menu Icon" width={70} height={70}
+				/>
+			</button>
+		</li>
 	</ul>
+	<div popover id="mobile-menu">
```

Comment:
Let's move this element outside of the `nav`. The way the structure is right now, two of the same menu are _nested_, so a keyboard or screen reader user would see something like this:
```
Main navigation menu
  - Toronto JS Logo!
  - Accessibility
  - License
  - About
  - Button: open blog navigation
  
Button: close blog navigation
  - Accessibility
  - License
  - About
```

A better way to have the menus would be:
```html
<nav aria-label="Main blog navigation" id="desktop-navigation">
  <!-- Desktop menu links -->
</nav>
<div id="mobile-navigation">
  <button type="button" id="nav-menu-button" aria-controls="mobile-menu" aria-expanded="false" aria-haspopup="true" aria-label="Open blog navigation" popovertarget="mobile-menu">
    <Icon name="icon reference" title="alt text for icon" />
  </button>
  <nav aria-label="Main blog navigation">
    <!-- The close button -->
    <!-- Mobile menu links, the same as desktop -->
  </nav>
</div>
```

Then with CSS we conditionally display and hide one of the two, either `#mobile-navigation` or `#desktop-navigation`.